SampleAPI/Controllers/EmployeeController.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using SampleAPI.Constant;
+using SampleAPI.Models;
+using SampleAPI.Repository;
+
+namespace SampleAPI.Controllers
+{
+    /// <summary>
+    /// EmployeeModel controller is used to perform CRUD operations with employee entity.
+    /// </summary>
+    [Route("sampleAPI/[controller]")]
+    [ApiController]
+    public class EmployeeController : ControllerBase
+    {
+        private readonly ILogger<EmployeeController> _logger;
+        private readonly IEmployeeRepository _repository;
+        public EmployeeController(ILogger<EmployeeController> logger, IEmployeeRepository repository)
+        {
+            _logger = logger;
+            _repository = repository;
+        }
+
+        /// <summary>
+        /// Get all employee records.
+        /// </summary>
+        /// <returns>Returns the employee list.</returns>
+        [HttpGet]
+        [Route("GetAllEmployee")]
+        public async Task<IActionResult> GetEmployee(string? search, string? filterByRole, bool includeInActive, [FromQuery] PagingParameterModel pagingparametermodel)
+        {
+            try
+            {
+                var response = new GetResponseModel<EmployeeModel>();
+                var result = await _repository.GetAllAsync(search, filterByRole, includeInActive);
+
+                int TotalPages = 0;
```

Comment:
This is not an efficient way to do pagination, we get all the records and then do in memory pagination. Please refer [this](https://stackoverflow.com/questions/7767409/better-way-to-query-a-page-of-data-and-get-total-count-in-entity-framework-4-1)

Reply:
Updated pagination logic as we discussed.